vivarium/core/serialize.py

```diff
@@ -185,99 +181,76 @@ def deserialize(  # pylint: disable=arguments-differ
                 unit_data.to(unit)
         return unit_data
 
+class QuantitySerializer(Serializer):
+    """Serializes data with units into strings of the form ``!units[...]``,
+    where ``...`` is the result of calling ``str(data)``. Deserializes strings
+    of this form back into data with units."""
+    python_type = type(1*units.fg)
 
-class NumpySerializer(Serializer):
-    """Serializer for Numpy arrays.
-    Numpy array serialization is lossy--we serialize to Python lists, so
-    deserialization will produce a Python list instead of a Numpy array.
-    """
-
-    class Codec(TypeEncoder):
-        python_type = np.ndarray
-        def transform_python(self, value: np.ndarray) -> List:
-            return value.tolist()
-
-    def get_codecs(self) -> List:
-        return [self.Codec()]
-
-class NumpyBoolSerializer(Serializer):
-    """Serializer for ``np.bool_`` objects."""
-    class Codec(TypeEncoder):
-        python_type = np.bool_
-        def transform_python(self, value: np.bool_) -> bool:
-            return bool(value)
-
-    def get_codecs(self) -> List:
-        return [self.Codec()]
-
-class NumpyInt64Serializer(Serializer):
```

Comment:
What happens to these different numpy types now?

Reply:
Orjson natively supports Numpy arrays and types, with the caveat of `np.str_` as noted in my initial comment.

Reply:
ah yes. that's a nice bonus improvement!